app/javascript/modules/listingDetailsLottery/ListingDetailsLotteryPreferencesEducator.tsx

```diff
@@ -0,0 +1,195 @@
+      {numVeteranApps ? (
+        <>
+          <p className="text-gray-700 text-sm">
+            {t("lottery.numberApplicantsQualifiedForPreference.veteran", {
+              number: numVeteranApps,
+              plural: numVeteranApps > 1 ? "s" : "",
+            })}
+          </p>
+          <p className="text-gray-700 text-sm">
+            {t("lottery.numberApplicantsQualifiedForPreference.nonVeteran", {
+              number: totalSubmittedApps,
+              plural: numVeteranApps > 1 ? "s" : "",
+            })}
+          </p>
+        </>
+      ) : (
+        <p className="text-gray-700 text-sm">
+          {t("lottery.numberApplicantsQualifiedForPreference", {
+            number: totalSubmittedApps,
+            plural: numVeteranApps > 1 ? "s" : "",
+          })}
+        </p>
+      )}
+    </div>
+    <hr className={lastPref ? "mt-4" : "my-4"} />
+  </div>
+)
+
+export const ListingDetailsLotteryPreferencesEducator = ({
+  lotteryBucketsDetails,
+  isEducatorOne,
+}: ListingDetailsLotteryPreferencesProps) => {
+  const veteranAppsByPreferenceMap = {}
+  lotteryBucketsDetails.lotteryBuckets
+    .filter((lotteryBucket) => preferenceNameHasVeteran(lotteryBucket.preferenceName))
+    .forEach((lotteryBucket) => {
+      veteranAppsByPreferenceMap[lotteryBucket.preferenceShortCode] =
+        lotteryBucket.totalSubmittedApps
+    })
+
+  return (
+    <div className="text-sm">
+      <header className="border-b border-gray-450 mb-4">
+        <Heading styleType="underlineWeighted" className="mx-8" priority={2}>
+          {t("lottery.housingPreferences.educator")}
+        </Heading>
+        <div className="pb-4 text-gray-700 text-xs mx-8 translate">
+          {renderMarkup(
+            `${t("lottery.rankingOrderNote", {
+              lotteryRankingVideoUrl: "https://www.youtube.com/watch?v=4ZB35gagUl8",
+            })}`
+          )}
+        </div>
+        <div className="pb-4 text-gray-700 text-xs mx-8">
+          {t("lottery.rankingOrderNote.veterans")}
+        </div>
+      </header>
+      <header className="border-b border-gray-450 mb-4">
+        <Heading styleType="underlineWeighted" className="mx-8 text-xs" priority={2}>
+          {t("listings.lotteryPreference.sfusd.educators.title")}
+        </Heading>
+      </header>
+      {lotteryBucketsDetails.lotteryBuckets
+        .filter(
+          (bucket) =>
+            bucket.preferenceShortCode &&
+            bucket.preferenceShortCode.includes("T1") &&
+            !bucket.preferenceShortCode.includes("V") &&
+            bucket.totalSubmittedApps
+        )
+        .map((bucket, index, arr) => (
+          <LotteryPreferences
+            unitsAvailable={bucket.unitsAvailable}
+            totalSubmittedApps={bucket.totalSubmittedApps}
+            preferenceName={bucket.preferenceName.replace("Tier 1 ", "")}
+            numVeteranApps={
+              veteranAppsByPreferenceMap[bucket.preferenceShortCode.replace("-", "-V-")] ?? null
```

Comment:
question: Can you elaborate on how this `replace` gets the correct submitted application count?

Reply:
I added code comments

The veteranAppsByPreferenceMap is keyed by the _veteran_ preference short code, which we then want to look up when we get to the associated preference bucket. We can derive the key of the map by inserting 'V-' into the non-veteran preference short code